### PR TITLE
test: update tests to use `jest` for test framework

### PR DESCRIPTION
**Pull request by jasonlafferty**
It appears that the current tests we have around `stubby` do no run, so we don't know which tests work and which are broken. This Pull request is trying to:
- [x] remove the existing testing framework and setup
- [x] add a `jest` as the new test framework
- [x] add a `jest` setup to create an environment that allows us to test `stubby`
- [x] update existing tests to work with the new environment setup
- [x] update test so that they work and reflect the real work

### CODE REVIEW

**Comment by jasonlafferty** [comment]
@waltfy have gotten most the test suite working, unsure what the broken test should be testing, could you have a look and give a hint (also @bianca93 if you know what they are). Feel free to fix up the tests 👍.

**Comment by waltfy** [comment]
Nice work, will take a look soon.

File: setupTests.js [hunk 1]
```diff
@@ -10,6 +11,23 @@ global.StubbySchemaValidator = require('./modules/schema-validator');
 // Start mock
 const toString = Object.prototype.toString;
 Object.prototype.toString = jest.fn().mockImplementation(() => '[object Object]');
+const originalDocumentCreateElement = document.createElement;
+
+document.createElement = jest.fn().mockImplementation((tagName) => {
```
Comment (jasonlafferty) [inline setupTests.js]: 👍 good catch 🎣

Comment (waltfy) [reply setupTests.js]: Was a tough one.

**Pull request merged**